src/main/scala/io/applicative/datastore/util/reflection/ReflectionHelper.scala

```diff
@@ -168,7 +168,12 @@ private[datastore] trait ReflectionHelper extends DateTimeHelper {
         val fieldClassName = member.returnType.typeSymbol.fullName
         val fieldName = member.name.toString
         if (fieldName == "id") {
-          field.set(entity.getKey.getNameOrId)
+          println(fieldClassName)
+          if (fieldClassName == "io.applicative.datastore.Key") {
```

Comment:
Could you replace the hardcoded `"io.applicative.datastore.Key"` with a constant such as `private val KeyClassName = getClassName[Key]()`?

Reply:
Done 👍